src/scripts/clipperUI/animations/animationStrategy.ts

```diff
@@ -1,3 +1,5 @@
+import {SmartValue} from "../../communicator/smartValue";
```

Comment:
I'm about to go through this PR so I will report back. My initial thought is to move `state` up one component so that it persists across re-renders of a sub-component, and pass it as a prop down the different layers of the animation strategy code.